services/ble/test/TestGattClient.cpp

```diff
@@ -77,24 +78,23 @@ class GattClientCharacteristicTest
 
 TEST_F(GattClientCharacteristicTest, receives_valid_notification_should_notify_observers)
 {
-    EXPECT_CALL(gattUpdateObserver, NotificationReceived(infra::ByteRangeContentsEqual(infra::MakeStringByteRange("string"))));
-    operations.infra::Subject<services::GattClientStackUpdateObserver>::NotifyObservers([](auto& observer)
+    const auto expectedData = infra::MakeStringByteRange("string");
+
+    EXPECT_CALL(gattUpdateObserver, NotificationReceived(infra::ByteRangeContentsEqual(expectedData)));
+    operations.infra::Subject<services::GattClientStackUpdateObserver>::NotifyObservers([&expectedData](auto& observer)
         {
-            observer.NotificationReceived(characteristicValueHandle, infra::MakeStringByteRange("string"));
+            observer.NotificationReceived(characteristicValueHandle, expectedData);
```

Comment:
If you use the value here, then please change its name: It's not merely _expected_ data, it is the actual data to be used. "data" is more appropriate as name. Or just hardcode "string" here.